pkg/apis/jaegertracing/v1/jaeger_types.go

```diff
@@ -207,13 +207,30 @@ type JaegerIngressSpec struct {
 	// +optional
 	OpenShift JaegerIngressOpenShiftSpec `json:"openshift,omitempty"`
 
+	// +optional
+	Hosts []string `json:"hosts,omitempty"`
+
+	// +optional
+	TLS []JaegerIngressTLSSpec `json:"tls,omitempty"`
```

Comment:
Wouldn't it be better to just use `Ingress` instead?

Reply:
Didn't quite get what you mean. Could you elaborate?

Reply:
Sorry, I meant that we could consider adding a new field holding `extv1beta1.Ingress` instead. We would still decorate the ingress with our own annotations, and we would probably adjust the backend values, but the ingress itself would be whatever the user is defining.

Note that I'm still not sure this is a good idea :-)

Reply:
I don't think that it's a good idea. That would:
1. Allow the user to specify values that will **always** be overridden / ignored (`rules[*].http.paths[*].path`, `rules[*].http.paths[*].backend.serviceName` and `backend.serviceName`);
2. Introduce additional error-prone matching / validation logic;
3. Potentially allow the user to expose non-jaeger services via jaeger operator's managed ingress.

Consider the following example:
```
metadata:
  name: "my-jaeger"
spec:
  strategy: allInOne
  allInOne:
    options:
      query:
        base-path: /jaeger # base-path is set here
  ingress: 
    rules:
      - host: jaeger.example.com
        http:
          paths:
            # should this be overridden based on the value above? just validated?
            - path: /query 
              backend:
                # do we validate this or overwrite with the generated service name?
                serviceName: my-jaeger-query
            # what about this?
            - path: /also-query 
              backend:
                serviceName: my-jaeger-query
      - host: also-jaeger.example.com
        http:
          paths:
            - backend:
                # should this even be allowed?
                serviceName: some-other-non-jaeger-service
```

Reply:
You are right, let's keep it simple for now.